### PR TITLE
doc: Add CNI chaining documentation for Kube-OVN

### PR DESCRIPTION
**Pull request by oilbeater**
Document how to use the generic-veth chaining plugin in combination with Kube-OVN.

Signed-off-by: oilbeater <liumengxinfly@gmail.com>

Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Thanks for contributing!

### CODE REVIEW

File: Documentation/gettingstarted/cni-chaining-kube-ovn.rst [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+   kubectl apply -f chaining.yaml
+
+
+Change Kube-OVN default install options
+=======================================
+
+To enable Cilium chaining mode and related functions, when install Kube-OVN the
+overlapped security function should be disabled and the cni conf priority should
+be modified.
```
Comment (qmonnet) [inline Documentation/gettingstarted/cni-chaining-kube-ovn.rst]: Would be nice to have a sentence describing why this changes are necessary

File: Documentation/gettingstarted/cni-chaining-kube-ovn.rst [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      --set cni.customConf=true \\
+      --set cni.configMap=cni-configuration \\
+      --set tunnel=disabled \\
+      --set enableIPv4Masquerade=false \\
+      --set enableIdentityMark=false
+
+.. note::
+
+   The new CNI chaining configuration will *not* apply to any pod that is
+   already running the cluster. Existing pods will be reachable and Cilium will
```
Comment (qmonnet) [inline Documentation/gettingstarted/cni-chaining-kube-ovn.rst]:
```suggestion
   already running in the cluster. Existing pods will be reachable and Cilium will
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
I'm not familiar with CNI chaining, but the doc looks good, thanks a lot!

Two minor comments below.

File: Documentation/gettingstarted/cni-chaining-kube-ovn.rst (hunk 1 above)
Comment (oilbeater) [reply Documentation/gettingstarted/cni-chaining-kube-ovn.rst]: add description why these options are needed

File: Documentation/gettingstarted/cni-chaining-kube-ovn.rst [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+.. code-block:: shell-session
+
+   kubectl apply -f chaining.yaml
+
+
+Change Kube-OVN default install options
+=======================================
+
+The Kube-OVN default installation option will implement the networkpolicy by it self,
```
Comment (qmonnet) [inline Documentation/gettingstarted/cni-chaining-kube-ovn.rst]:
```suggestion
The Kube-OVN default installation option will implement the networkpolicy by itself,
```

**Review by qmonnet — APPROVED** [review]
Looks good, thanks!

One question that comes up though, can you please explain what `--set enableIdentityMark=false` does? I'm not familiar with the option

**Comment by oilbeater** [comment]
> can you please explain what --set enableIdentityMark=false does? I'm not familiar with the option

This option is not documented now, but from the source code here:

https://github.com/cilium/cilium/blob/master/pkg/option/config.go#L308

Without this option, Cilium will add mark to traffic that can be conflict with Kube-OVN mark

File: Documentation/gettingstarted/cni-chaining-kube-ovn.rst (hunk 3 above)
Comment (oilbeater) [reply Documentation/gettingstarted/cni-chaining-kube-ovn.rst]: done

**Comment by oilbeater** [comment]
Hi, any update on this？

**Comment by joestringer** [comment]
I'll mark this as draft until I see feedback on the latest comment above. Feel free to mark it as "ready for review again" and ping me when there is an update. Thanks!
